SD_converter.py

```diff
@@ -0,0 +1,136 @@
+import json
+import os
+from typing import List, Tuple
+
+
+class Converter:
+    """A class to represent a converter for AI fine-tuning notations."""
+
+    def __init__(self):
+        return
+
+    def _to_file(self, output_file: str, content: List[str]) -> None:
+        """
+        Writes the content to a target file.
+
+        Args:
+            output_file (str): Target file.
+            content (List[str]): Content, usually .jsonl format written in lines.
+        """
+        with open(output_file, "w") as f:
+            for line in content:
+                f.write(f"{line}\n")
+
+    def _parse_input_file(self, input_file: str) -> List[Tuple[str, str]]:
+        """
+        Reads and parses the input file into a list of Question-Answer pairs.
+
+        Args:
+            input_file (str): Path to the input file.
+
+        Returns:
+            List[Tuple[str, str]]: A list of Question-Answer pairs.
+        """
+
+        try:
+            with open(input_file, "r") as file:
+                text = file.read()
+            entries = text.split("Question:")
+
+            if len(entries) == 1:
+                raise ValueError(f"No 'Question:' found in the input file: {input_file}")
+
+            qa_pairs = []
+            for entry in entries[1:]:
+                try:
+                    question, answer = entry.split("Answer:")
+                except ValueError:
+                    raise ValueError(f"Missing 'Answer:' keyword in the input file: {input_file}")
+                question = question.strip()
+                answer = answer.strip()
+                qa_pairs.append((answer, question))
+
+        except FileNotFoundError:
+            raise FileNotFoundError(f"File not found: {input_file}")
+        except Exception:
+            raise Exception(f"Invalid file format: {input_file}")
+
+        return qa_pairs
+
+    def _to_oai(self, qa_pairs: List[Tuple[str, str]]) -> List[str]:
+        """
+        Converts a list of Question-Answer pairs to the JSONL format and writes to the output file.
+
+        Args:
+            qa_pairs (List[Tuple[str, str]]): A list of Question-Answer pairs.
+
+        Returns:
+            List[str]: Content of the final .jsonl file
+        """
+
+        content = []
+        for answer, question in qa_pairs:
+            qa_pair = {"question": question, "answer": answer}
```

Comment:
Not correct. See [openai documentation](https://platform.openai.com/docs/guides/fine-tuning/prepare-training-data).

`{"prompt": "<prompt text>", "completion": "<ideal generated text>"}`